src/nl_params.jl

```diff
@@ -85,3 +85,9 @@ const nary_functions = Set{Symbol}([
 # :or_n,
 # :alldiff,
 ])
+
+const POSSIBLE_USERLIMITS = [
+    "iterations exceeded",  # Max iterations
+    "acceptable level",     # Acceptable tolerance/iterations
```

Comment:
The acceptable level one? 

"if the algorithm encounters "acceptable_iter" many iterations in a row that are considered "acceptable", it will terminate before the desired convergence tolerance is met. This is useful in cases where the algorithm might not be able to achieve the "desired" level of accuracy"

Maybe it should return `Optimal`...